services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -0,0 +1,79 @@
+package cromwell.pipeline.service
+
+import java.net.URLEncoder
+import java.nio.file.Paths
+
+import akka.http.scaladsl.model.StatusCodes
+import cromwell.pipeline.datastorage.dao.repository.utils.TestProjectUtils
+import cromwell.pipeline.datastorage.dto.ProjectFile
+import org.mockito.Mockito.when
+import org.scalatest.{ AsyncWordSpec, Matchers }
+import org.scalatestplus.mockito.MockitoSugar
+import play.api.libs.json.Json
+
+import scala.concurrent.Future
+
+class GitLabProjectVersioningTest extends AsyncWordSpec with Matchers with MockitoSugar {
+  private val httpClient = mock[HttpClient]
+  private val gitLabProjectVersioning = new GitLabProjectVersioning(httpClient)
+  private val dummyProject = TestProjectUtils.getDummyProject()
+
+  import ProjectFileContext._
+
+  "GitLabProjectVersioningTest" when {
+
+    "updateFile" should {
+      "return create file response when file is new" taggedAs Service in {
+        val path = URLEncoder.encode(newFile.path.toString, "UTF-8")
+
+        when(
+          httpClient.put(
+            s"https://gitlab.com/api/v4/projects/${dummyProject.projectId}/repository/files/$path",
+            payload = Json.stringify(Json.toJson(Map("branch" -> "master", "content" -> newFile.content))),
+            headers = Map("Private-Token" -> "your_token")
+          )
+        ).thenReturn(
+          Future.successful(
+            Response(StatusCodes.BadRequest.intValue, "File does not exist", Map("error" -> List("true")))
+          )
+        )
+        when(
+          httpClient.post(
+            s"https://gitlab.com/api/v4/projects/${dummyProject.projectId}/repository/files/$path",
+            payload = Json.stringify(
+              Json.toJson(Map("branch" -> "master", "content" -> newFile.content, "commit_message" -> "Init commit"))
+            ),
+            headers = Map("PRIVATE-TOKEN" -> "your_token")
```

Comment:
Use val for this. (headers = Map("PRIVATE-TOKEN" -> "your_token")) We will use it many times.

Reply:
You are right